### PR TITLE
Add new invoicing page

### PR DESCRIPTION
**Pull request by aanil**
Added new invoicing page
Added new `PC Admin` option on toolbar.
Added new `proj_coord` role

Up on stage!

### CODE REVIEW

File: conda_requirements.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#name: genstat
```
Comment (aanil) [inline conda_requirements.yml:1]:
```suggestion
name: genstat
```

**Review by alneberg — CHANGES_REQUESTED** [review]
This will be amazing! I haven't gotten to read through the python file yet, will do tomorrow. Some suggested changes to the UI, but most of it looks very nice!

File: run_dir/design/invoicing.html [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            </div>
+            <table class="table table-hover table-striped table-bordered sortable" id="invoicing_table">
+              <thead>
+                <tr class="sticky darkth">
+                  <th class="mw-5 text-center">
+                      <input class="form-check-input" type="checkbox" value="" id="sel_all_invoices">
+                  </th>
+                  <th class="sort">Project</th>
+                  <th class="sort">Date of Generation</th>
+                  <th class="sort">View Invoice</th>
```
[the remark this answers is not in the document]
Comment (aanil) [reply run_dir/design/invoicing.html]: Done!

File: run_dir/design/invoicing.html [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        </div>
+        <div class="tab-pane fade" id="tab_sent_invoices">
+          <div id="sent_invoices_table_div">
+              <h2><span id="sent_invoices_tab_title">Sent invoices</span>
+                <small class="text-muted">Showing projects that have had invoice sent in the last 6 months</small>
+              </h2>
+              <table class="table table-hover table-striped table-bordered sortable" id="sent_invoices_table">
+                <thead>
+                  <tr class="sticky darkth">
+                    <th class="sort">Date of Generation</th>
```
[the remark this answers is not in the document]
Comment (aanil) [reply run_dir/design/invoicing.html]: Done!

File: run_dir/static/js/invoicing.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+          success: function(data, textStatus, xhr) {
+          $('#view_invoice').append(data);
+          }
+        });
+      });
+    })
+  }
+
+  // Initialize sorting and searching javascript plugin
+  function init_listjs(table_name) {
```
[the remark this answers is not in the document]
Comment (aanil) [reply run_dir/static/js/invoicing.js:50]: Done( for existing tables too)

File: run_dir/static/js/invoicing.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  $('#invoicing_generate_invoice_btn').click(function(e){
+    var projs = []
+    $('.invoice_checkbox').each(function(){
+      if($(this).prop('checked')===true){
+        projs.push($(this).prop('value'))
+      }
+    })
+    $("#sel_proj_form").remove()
+    $("#gen_invoice_form").append("<input type='hidden' id='sel_proj_form' name='projects' value="+projs+">")
+    $('#gen_invoice_form').attr('action', '/generate_invoice').attr('method', 'post').delay(3000).queue(function(){load_invoicing_table()})
```
[the remark this answers is not in the document]
Comment (aanil) [reply run_dir/static/js/invoicing.js:99]: The request submits to a new page, and we can't actually have a promise resolve between pages, I think. So I just refresh the table after 3 seconds.

File: run_dir/static/js/pricing_quote.js [hunk 6]
```diff
@@ -208,6 +210,23 @@ app.component('v-pricing-quote', {
             })
           }
         },
+        generate_invoice_spec(){
```
[the remark this answers is not in the document]
Comment (aanil) [reply run_dir/static/js/pricing_quote.js:213]: Done!

**Comment by alneberg** [comment]
I think we need to figure out a way for the "Mark signed"/"spec generated" to not be untraceably changed after the invoice has been downloaded. I guess we're not supposed to do that but I find it a bit troublesome that if it's done, one won't even know.

**Comment by aanil** [comment]
> I think we need to figure out a way for the "Mark signed"/"spec generated" to not be untraceably changed after the invoice has been downloaded. I guess we're not supposed to do that but I find it a bit troublesome that if it's done, one won't even know.

Change the field to a list?

**Review by alneberg — APPROVED** [review]
Woohoo! Let's go!

**Pull request merged**